src/components/LoginHeader.tsx

```diff
@@ -0,0 +1,49 @@
+import { RouteProp } from '@react-navigation/native';
+import React, { useContext } from 'react';
+import { TouchableOpacity, ViewStyle } from 'react-native';
+
+import { SettingsContext } from '../SettingsProvider';
+
+import { Icon, colors, consts } from './../config';
+import { useLoginProfile } from './../hooks';
+
+export const LOGIN_MODAL = 'loginModal';
+
+const { a11yLabel } = consts;
+
+type Props = {
+  route: RouteProp<any, string>;
+  style: ViewStyle;
+};
+
+export const LoginHeader = ({ route, style }: Props) => {
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings } = globalSettings;
+  const { hasProfile = false } = settings;
+
+  const { isLoggedIn, loading, login, logout } = useLoginProfile();
+
+  if (!hasProfile || loading) {
+    return null;
+  }
```

Comment:
checking `hasProfile` should not happen here but already considered in the stack config boolean flag `withLogin`